src/index.ts

```diff
@@ -149,46 +181,58 @@ app.get('/repodetails/:loginName', authenticateToken, async (req: Request, res:
 });
 
 
-app.get('/runnertoken/:org/:reponame', authenticateToken,  async (req:Request, res:Response) => {
+app.get('/runnertoken/:org/:reponame/:accesstoken',  async (req:Request, res:Response) => {
     const ORG = req.params.org;
     const REPONAME = req.params.reponame;
 
-    // initalize octokit
-    const instance = appManager.getOctokitInstance(ORG);
-    var octo:any = null;
-    var installationID: number = 0;
-    if (instance.length != 0){
-        installationID = instance[0].id;
-        octo = instance[0].octokit;
+    // check if access token is correct
+    const tokenMap = appManager.getAccessTokens()
+    if (tokenMap[ORG+" "+REPONAME] == req.params.accesstoken){
```

Comment:
If there is a request from GitHub Actions to create, say, 2 runners, then it looks like this code will only allow one of them through, because the code is checking the validity of the token by checking to see if it is the most recent token created. Instead, can this be changed to validate the token by verifying and decoding the token (using the repo name that is encoded in the token, not in the request)?

Reply:
Added code to decode from token